src/ConcurrentIteratorChannel.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Amp\Sync;
+
+use Amp\ByteStream\ClosableStream;
+use Amp\Cancellation;
+use Amp\Pipeline\ConcurrentIterator;
+use Amp\Pipeline\Queue;
+
+/**
+ * Creates a channel from a Pipeline and Emitter. The Pipeline emits data to be received on the channel (data
+ * emitted on the Pipeline will be returned from calls to {@see Channel::receive()}). The Emitter will receive data
+ * that sent on the channel (data passed to {@see Channel::send()} will be passed to {@see Emitter::yield()}).
+ *
+ * @template TReceive
+ * @template TSend
+ * @template-implements Channel<TReceive, TSend>
+ */
+final class ConcurrentIteratorChannel implements Channel, ClosableStream
```

Comment:
Should `Channel` simply have a `close` method itself instead of using `ClosableStream`?

Reply:
I think that makes more sense, yes. Updated.